packages/react-native-app-auth/ios/RNAppAuth.m

```diff
@@ -499,35 +538,54 @@ - (void)endSessionWithConfiguration: (OIDServiceConfiguration *) configuration
                                     postLogoutRedirectURL: [NSURL URLWithString:postLogoutRedirectURL]
                                      additionalParameters: additionalParameters];
 
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
     id<UIApplicationDelegate, RNAppAuthAuthorizationFlowManager> appDelegate = (id<UIApplicationDelegate, RNAppAuthAuthorizationFlowManager>)[UIApplication sharedApplication].delegate;
+#elif TARGET_OS_OSX
+    id<NSApplicationDelegate, RNAppAuthAuthorizationFlowManager> appDelegate = (id<NSApplicationDelegate, RNAppAuthAuthorizationFlowManager>)[NSApplication sharedApplication].delegate;
+#endif
     if (![[appDelegate class] conformsToProtocol:@protocol(RNAppAuthAuthorizationFlowManager)]) {
         [NSException raise:@"RNAppAuth Missing protocol conformance"
                     format:@"%@ does not conform to RNAppAuthAuthorizationFlowManager", appDelegate];
     }
     appDelegate.authorizationFlowManagerDelegate = self;
     __weak typeof(self) weakSelf = self;
 
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
     rnAppAuthTaskId = [UIApplication.sharedApplication beginBackgroundTaskWithExpirationHandler:^{
         [UIApplication.sharedApplication endBackgroundTask:rnAppAuthTaskId];
         rnAppAuthTaskId = UIBackgroundTaskInvalid;
     }];
 
     UIViewController *presentingViewController = appDelegate.window.rootViewController.view.window ? appDelegate.window.rootViewController : appDelegate.window.rootViewController.presentedViewController;
+#elif TARGET_OS_OSX
+    NSWindow *presentingWindow = [NSApplication sharedApplication].mainWindow;
+    if(!presentingWindow){
+        reject(@"authentication_failed", @"Unable to get the main window to present an authorization request. Try authenticating again with the main window open.", nil);
+        return;
+    }
+#endif
 
 #if TARGET_OS_MACCATALYST
     id<OIDExternalUserAgent> externalUserAgent = nil;
 #elif TARGET_OS_IOS
     id<OIDExternalUserAgent> externalUserAgent = iosCustomBrowser != nil ? [self getCustomBrowser: iosCustomBrowser] : [self getExternalUserAgentWithPresentingViewController:presentingViewController
                                                                                                                                     prefersEphemeralSession:prefersEphemeralSession];
+#elif TARGET_OS_OSX
+    id<OIDExternalUserAgent> externalUserAgent = nil;
```

Comment:
I was unsure about this `externalUserAgent = nil`, but it does turn out that under-the-hood for the APIs we're calling, the AppAuth library makes an appropriate `OIDExternalUserAgentMac` for you so there's no need to make one yourself.